### PR TITLE
fix: fix_sort_null_panic

### LINKED ISSUE #12718: bug: `array_sort` function panic if the inner type is Null or EmptyArray
### Search before asking

- [X] I had searched in the [issues](https://github.com/datafuselabs/databend/issues) and found no similar issues.

### Version

v1.2.100-nightly

### What's Wrong?

`array_sort` function panic if the inner type is Null or EmptyArray
```
panicked at 'called `Result::unwrap()` on an `Err` value: FromStdError. Code: 1001, Text = Invalid argument error: The data type type Null has no natural order.
panicked at 'called `Result::unwrap()` on an `Err` value: Unimplemented. Code: 1002, Text = arrow: Sort not supported for data type Extension("EmptyArray", Null, None).
```

### How to Reproduce?

```sql
mysql> select array_sort([[], []]);
ERROR 2013 (HY000): Lost connection to MySQL server during query
No connection. Trying to reconnect...
Connection id:    10
Current database: *** NONE ***


mysql> select array_sort([null, null]);
ERROR 2013 (HY000): Lost connection to MySQL server during query
No connection. Trying to reconnect...
Connection id:    14
Current database: *** NONE ***
```

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by ct20000901**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

When the sorted array is an empty array or the sorted content is null, DataBlock::sort() will return Err. In such cases, unwrapping directly is not possible and requires handling.

- Closes #12718

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/12827)

### CODE REVIEW

File: src/query/functions/src/scalars/array.rs [hunk 1]
```diff
@@ -751,10 +751,12 @@ fn register_array_aggr(registry: &mut FunctionRegistry) {
                     }];
                     let columns = vec![BlockEntry{
                         data_type: arr.data_type(),
-                        value: Value::Column(arr)
+                        value: Value::Column(arr.clone()),
                     }];
-                    let sort_block = DataBlock::sort(&DataBlock::new(columns, len), &sort_desc, None).unwrap();
-                    sort_block.columns()[0].value.clone().into_column().unwrap()
+                    match DataBlock::sort(&DataBlock::new(columns, len), &sort_desc, None) {
```
Comment (b41sh) [inline src/query/functions/src/scalars/array.rs:756]: Ignoring errors may lead to unintended results. We need to implement sort for `Null`, `EmptyArray`, and `EmptyMap` types in `build_compare` function. Add a custom compare function for those types, for example
```rust
fn compare_null() -> ArrowResult<DynComparator> {
    Ok(Box::new(move |_, _| Ordering::Equal))
}
```

Comment (ct20000901) [reply src/query/functions/src/scalars/array.rs]: i have completed this. Please review this pr again. Thank you very much.

File: src/query/expression/src/kernels/sort.rs [hunk 2]
```diff
@@ -239,16 +244,15 @@ fn compare_decimal256(left: &dyn Array, right: &dyn Array) -> ArrowResult<DynCom
...
+            "EmptyArray" => compare_null(left, right),
+            _ => Err(ArrowError::NotYetImplemented(format!(
+                "Sort not supported for data type {:?}",
+                left.data_type()
+            ))),
+        },
+        ArrowType::Null => compare_null(left, right),
         ArrowType::Decimal256(_, _) => compare_decimal256(left, right),
         _ => arrow_ord::build_compare(left, right),
     }
```
Comment (b41sh) [inline src/query/expression/src/kernels/sort.rs:258]: Please add some unit tests and SQL logic tests for those cases.
You can refer to the following examples:
https://github.com/datafuselabs/databend/blob/main/src/query/functions/tests/it/scalars/array.rs#L646
https://github.com/datafuselabs/databend/blob/main/tests/sqllogictests/suites/query/02_function/02_0061_function_array#L177

Comment (ct20000901) [reply src/query/expression/src/kernels/sort.rs:258]: > Please add some unit tests and SQL logic tests for those cases. You can refer to the following examples: https://github.com/datafuselabs/databend/blob/main/src/query/functions/tests/it/scalars/array.rs#L646 https://github.com/datafuselabs/databend/blob/main/tests/sqllogictests/suites/query/02_function/02_0061_function_array#L177

I have added the corresponding test case for this change, but I am unsure if it complies with databend's testing specifications. If there is anything incorrect, I would appreciate it if you could point it out. Thank you

**Review by b41sh — APPROVED** [review]

**Comment by b41sh** [comment]
LGTM, thanks for your contribution @ct20000901

**Pull request merged**